### PR TITLE
[Mac] implement chirality

### PR DESCRIPTION
**Pull request by tombogle**
This solves issue #178

### CODE REVIEW

File: mac/KeymanEngine4Mac/KeymanEngine4MacTests/KMEngineTests.m [hunk 1]
```diff
@@ -12,10 +12,12 @@
 #import "KMEngine.h"
 
 @interface KMEngineTests : XCTestCase
-
 @end
 
 @implementation KMEngineTests
+const int nCombinations = 14;
```
Comment (mcdurdin) [inline mac/KeymanEngine4Mac/KeymanEngine4MacTests/KMEngineTests.m]: Good to document why this many combinations :-)

File: mac/KeymanEngine4Mac/KeymanEngine4MacTests/KMEngineTests.m [hunk 2]
```diff
@@ -84,16 +94,179 @@ - (void)testprocessEvent_eventForLowercaseA_ReturnsQstrActionWithExpectedCharact
...
-//    XCTAssert(actions.count == 1, @"Expected 1 action");
-//    NSDictionary *action = actions[0];
-//    NSString *actionType = [[action allKeys] objectAtIndex:0];
-//    XCTAssert([actionType isEqualToString:Q_STR], @"Expected Q_STR action");
-//    NSString *output = [action objectForKey:actionType];
-//    XCTAssert([output isEqualToString:@"\u00A3"], @"Expected pound sign (U+00A3)");
-//}
++ (void)fillInNamesAndModifiersForAllChiralCombinations {
+    int i = 0;
+    modifiers[i] = LEFT_CTRL_FLAG;
```
Comment (mcdurdin) [inline mac/KeymanEngine4Mac/KeymanEngine4MacTests/KMEngineTests.m]: Given we have a fixed number of combinations, I'd be more comfortable using integer values than an incrementing index here -- it's cleaner to read and understand., e.g. `modifiers[0] = ...; modifiers[1] = ...;`

Comment (tombogle) [reply mac/KeymanEngine4Mac/KeymanEngine4MacTests/KMEngineTests.m]: Maybe so. The advantage of using an index that can be incremented is that it's fast for copying and pasting, especially if you ever stick another entry in somewhere besides the end. I really wanted to use a dictionary, but NSDictionary doesn't support non-object types as either keys or values.

Comment (mcdurdin) [reply mac/KeymanEngine4Mac/KeymanEngine4MacTests/KMEngineTests.m]: The disadvantage is that you can overflow bounds without noticing :)

**Pull request merged**